consensus.go

```diff
@@ -38,16 +39,25 @@ type RouterFlags struct {
 	V2Dir     bool
 }
 
+type RouterAddress struct {
+	IPv4Address net.IP
+	IPv4ORPort  uint16
+	IPv4DirPort uint16
+
+	IPv6Address net.IP
+	IPv6ORPort  uint16
```

Comment:
Is it possible to run a v4/v6 relay on two different ports? If so, this seems fine but otherwise one variable for the port would be more elegant.

Reply:
You can specify different ORPorts for v4 and v6, though it doesn't look common. 

https://collector.torproject.org/recent/relay-descriptors/consensuses/ is down right now, so I can't find an example from a recent consensus file.

```bash
From 2017-09-11 21:00:00's consensus file I see examples of them being different which is why I added it:
r havingtrouble AEJpGAnlKLJYyf4BKhxGhZFj8UI +lzGWxad1rGlBC8ZehFOSXnL2mI 2017-09-11 02:59:33 159.203.42.254 9001 443
a [2604:a880:cad:d0::862:4001]:9050

r pkswitchny Ap4QelnStP69BqYRUWe9bQSNnAo J8GszXnVclt6yHS9nNVOFb8VqRU 2017-09-11 16:13:27 104.236.52.16 1747 0
a [2604:a880:800:10::63b:c00a]:7491
```

Only documentation I can find says don't specify DirPort for ipv6 because it will be ignored
https://trac.torproject.org/projects/tor/wiki/doc/IPv6RelayHowto

Reply:
Maybe the answer was in front of me in the tor wiki link :).

" The way you make your relay bind to and publish an IPv6 OR port is to add another ORPort config option to your torrc file. You can keep your IPv4 ORPort option as is. "
"Your relay will have to publish both an IPv4 and an IPv6 OR port."